### PR TITLE
SOCKS5 over SSH Tunnel Support

### PR DESCRIPTION
**Pull request by ikreymer**
Adds support for running a SOCKS5 proxy over an SSH connection. This can be configured by using `--proxyServer ssh://user@host[:port]` config and also passing an `--sshProxyPrivateKeyFile <private key file>` file param and an optional `--sshProxyKnownHostsFile <public host key file>`file param. The key files are expected to be mounted as volumes into the crawler.

The proxy config uses autossh to establish a more robust connection, and also waits until a connection can be established before proceeding.

### CODE REVIEW

**Comment by vnznznz** [comment]
some additional tests would be great

File: src/util/proxy.ts [hunk 1]
```diff
@@ -58,3 +74,127 @@ export function createDispatcher(proxyUrl: string): Dispatcher | undefined {
...
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function runSSHD(params: Record<string, any>, detached: boolean) {
+  const { proxyServer } = params;
+  if (!proxyServer || !proxyServer.startsWith("ssh://")) {
+    return "";
+  }
+
+  const hostPort = proxyServer.slice(6).split(":");
```
[the remark this answers is not in the document]
Comment (ikreymer) [reply src/util/proxy.ts]: Switched to parsing URL. Note that this requires ipv6 addresses to be wrapped in `[]` but that seems to be standard enough.

**Comment by ikreymer** [comment]
Added an SSH tunnel test using a local [linuxserver/openssh-server](https://docs.linuxserver.io/images/docker-openssh-server/) to run crawl. (Could add an actual proxy later if needed, but this should be sufficient for now I think)

**Review by tw4l — APPROVED** [review]
Tested with root and non-root users, rsa and ed25519 keys, as well as with a hosts file. All working very well, nice work!

**Pull request merged**